### PR TITLE
Changed DirEntryExists and GetFileInfo to use std::filesystem on Windows since _stat is broken

### PR DESCRIPTION
**Pull request by drfrag666**

### CODE REVIEW

**Comment by coelckers** [comment]
This will only be added if you can conclusively prove that it fixes the problem.
I have doubts that the case this is supposed to hande will ever be triggered under normal circumstances.

**Comment by drfrag666** [comment]
Which problem? FileExists not working? I've confirmed it with the other PR, denying all permissions for the ini file you get an abort with the can't load ini error message with this, else it's like if the file didn't exist and factory defaults will be loaded and on exit windows will give an error saying it can't write the file. The doubt i have here is about the specification change for GetFileInfo (time is unused) and using the internal _stat instead but the transition to std::filesystem could be completed at a later stage.
If you're referring to the related issue of the dissapearing ini it's rare but it happens as it's been reported and i don't know what else could happen with a bad FileExists.

**Comment by coelckers** [comment]
I mean, if this doesn't solve the INI problem conclusively, I won't add it. The _stat bug you mentioned is an edge case that should never happen with any proper setup of the INI file.

File: src/common/utility/cmdlib.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ char(&_ArraySizeHelper(T(&array)[N]))[N];
 bool FileExists (const char *filename);
 bool DirExists(const char *filename);
 bool DirEntryExists (const char *pathname, bool *isdir = nullptr);
-bool GetFileInfo(const char* pathname, size_t* size, time_t* time);
+bool GetFileInfo(const char* pathname, size_t* size);
```
Comment (alexey-lysiuk) [inline src/common/utility/cmdlib.h:38]: Raze [uses](https://github.com/coelckers/Raze/blob/a00da236541fddc6f9c96ebfe410e681e3180918/source/core/searchpaths.cpp#L304) the third argument of `GetFileInfo()` function.

**Review by alexey-lysiuk — COMMENTED** [review]
If you claim that `_stat()` broken, where is the specification of its behavior in this malfunctioning case? I would like to get information about it from official sources.

**Comment by drfrag666** [comment]
There are two different issues, this is not to fix the ini issue (that's what the other PR is for). The ini issue can be easily reproduced removing all permissions from the ini and if you restore them before exiting you'll screw your ini, i investigated the source and now i've reproduced it doing this. It's due to the lack of a FileExists check.
This issue is about the FileExists check itself and it's way more general, when files are in use by something else. The _stat function pretends to be like stat and of course i checked the specification. https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/stat-functions?view=msvc-160
"Each of these functions returns 0 if the file-status information is obtained. A return value of -1 indicates an error, in which case errno is set to ENOENT, indicating that the filename or path could not be found. A return value of EINVAL indicates an invalid parameter; errno is also set to EINVAL in this case." With the internal _stat function FileExists says the file exists (it does), same for filesystem::exists. But now i'll have to use the internal  _stat with a _stat64 struct and that should work. @Blzut3 made it and originally was for XP. I don't think it's possible to do the conversion from file_time_type to time_t.

**Comment by alexey-lysiuk** [comment]
In the provided reference, I don't see any information about behavior of `_stat()` in case of inability to access a file.
At the same time, [POSIX specification](https://pubs.opengroup.org/onlinepubs/009696699/functions/stat.html) describes it clearly.
> Read, write, or execute permission of the named file is not required.

That's why I was asking why you decide that MSVC `_stat()` is broken.
Microsoft may or may not follow POSIX standard as Windows is certainly not POSIX compatible OS.

**Comment by drfrag666** [comment]
It clearly says "indicating that the filename or path could not be found" and it doesn't say that permissions are required so they are not. It's a bug and it's used to check for file existence everywhere but even if it wasn't that's not relevant, then _stat would not be adequate by spec.

**Comment by drfrag666** [comment]
So now is this right in _stat64?
`	buffer->st_size = ((uint64_t)data.nFileSizeHigh << 32) + data.nFileSizeLow;`

**Comment by drfrag666** [comment]
I'm closing this and pushed https://github.com/coelckers/gzdoom/pull/1238 instead.

**Pull request closed without merging**